ts/src/hashkey.ts

```diff
@@ -0,0 +1,4336 @@
+         * @param {string} [params.type] 'spot' or 'swap' - the type of the market to fetch entry for (default 'spot')
+         * @returns {object} an list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        const methodName = 'cancelOrders';
+        await this.loadMarkets ();
+        const request = {};
+        let orderIds = '';
+        for (let i = 0; i < ids.length; i++) {
+            orderIds += this.safeString (ids, i) + ',';
+        }
+        orderIds = orderIds.slice (0, -1);
+        request['ids'] = orderIds;
+        let market: Market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        let marketType = 'spot';
+        [ marketType, params ] = this.handleMarketTypeAndParams (methodName, market, params, marketType);
+        if (marketType === 'spot') {
+            return await this.privateDeleteApiV1SpotCancelOrderByIds (this.extend (request));
+            //
+            //     {
+            //         "code": "0000",
+            //         "result": []
+            //     }
+            //
+        } else if (marketType === 'swap') {
+            return await this.privateDeleteApiV1FuturesCancelOrderByIds (this.extend (request));
+        } else {
+            throw new NotSupported (this.id + ' ' + methodName + '() is not supported for ' + marketType + ' type of markets');
+        }
+    }
+
+    async fetchOrder (id: string, symbol: Str = undefined, params = {}): Promise<Order> {
+        /**
+         * @method
+         * @name hashkey#fetchOrder
+         * @description fetches information on an order made by the user
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/query-order
+         * @param {string} id the order id
+         * @param {string} symbol unified symbol of the market the order was made in
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.type] 'spot' or 'swap' - the type of the market to fetch entry for (default 'spot')
+         * @param {string} [params.clientOrderId] a unique id for the order that can be used as an alternative for the id
+         * @param {string} [params.accountId] *spot markets only* account id to fetch the order from
+         * @param {bool} [params.trigger] *swap markets only* true for fetching a trigger order (default false)
+         * @param {bool} [params.stop] *swap markets only* an alternative for trigger param
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        const methodName = 'fetchOrder';
+        this.checkTypeParam (methodName, params);
+        await this.loadMarkets ();
+        const request: Dict = {};
+        let clientOrderId: Str = undefined;
+        [ clientOrderId, params ] = this.handleParamString (params, 'clientOrderId');
+        if (id !== undefined) {
```

Comment:
due to the method's signature, `id` is always expected to be defined, so we should first check if clientOrderid is not undefined, and then use the id

Reply:
done